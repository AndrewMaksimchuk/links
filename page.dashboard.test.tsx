import { describe, test, expect } from 'bun:test'
import { renderToReadableStream } from 'hono/jsx/streaming'
import { Dashboard } from './page.dashboard';
import type { User } from './service.user';
import type { Tag } from './service.tag';


const user: User = {
    name: 'Andrew',
    password: 'root',
    telephone: '+380',
    user_id: 0,
}


const tag: Tag = {
    color: 'blue',
    name: 'web',
    tag_id: 0,
}


describe('Page Dashboard', () => {
    test('view card', async () => {
        const stream = renderToReadableStream(<Dashboard
            linkViewState='card'
            tags={[]}
            user={user} />)
        expect(await Bun.readableStreamToText(stream)).toMatchSnapshot()
    })

    test('view table', async () => {
        const stream = renderToReadableStream(<Dashboard
            linkViewState='table'
            tags={[]}
            user={user} />)
        expect(await Bun.readableStreamToText(stream)).toMatchSnapshot()
    })


    test('view table with tags', async () => {
        const stream = renderToReadableStream(<Dashboard
            linkViewState='table'
            tags={[tag]}
            user={user} />)
        expect(await Bun.readableStreamToText(stream)).toMatchSnapshot()
    })
})
