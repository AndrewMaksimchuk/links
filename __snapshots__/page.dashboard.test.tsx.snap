// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Dashboard view card 1`] = `
"<header><div style="display:flex;justify-content:space-between;align-items:center;margin-bottom:var(--pico-typography-spacing-vertical)"><h1 id="userGreeting" style="margin: 0;"><a href="/dashboard">Hello, admin!</a></h1><div style="display: flex; align-items: center; gap: 1rem;"><label><span>Dark</span><input type="checkbox" role="switch" style="margin-inline: .5em;" x-data="{
        dataTheme: undefined, 
        dataThemeValues: [&#39;dark&#39;, &#39;light&#39;],
        get dataThemeInputValue() {
            return &#39;dark&#39; === this.dataTheme ? false : true; 
        },
        setThemeCookie() {
            document.cookie = &#39;theme=&#39; + this.dataTheme + &#39;;SameSite=Strict;&#39;;
        },
        getThemeCookie() {
            const theme = document.cookie?.includes(&#39;theme&#39;) ? document.cookie.split(&#39;=&#39;).at(1) : undefined;
            return this.dataThemeValues.includes(theme) ? theme : undefined;
        },
        setTheme() {
            this.setThemeCookie();
            document.documentElement.setAttribute(&#39;data-theme&#39;, this.dataTheme);
        },
        toggleTheme() {
            this.dataTheme = &#39;dark&#39; === this.dataTheme ? &#39;light&#39; : &#39;dark&#39;;
            this.setTheme();
        },
            init() {
                const cookieTheme = this.getThemeCookie();
                if(cookieTheme) {
                    this.dataTheme = cookieTheme;
                    this.setTheme();
                    return;
                }
                this.dataTheme = window.matchMedia(&#39;(prefers-color-scheme: dark)&#39;).matches ? &#39;dark&#39; : &#39;light&#39;;
            },
    }" x-on:click="toggleTheme" x-model="dataThemeInputValue"/><span>Light</span></label><button class="outline secondary" onclick="settingsView.toggleAttribute(&#39;open&#39;); document.body.classList.toggle(&#39;modal-is-open&#39;);history.pushState({}, &#39;&#39;, &#39;/settings&#39;);">Settings</button><form action="/logout" method="post" style="margin: 0;"><button type="submit" class="outline" style="margin: 0;">Logout</button></form></div></div><nav style="display: grid; grid-template-columns: 1fr;"><form id="linkAddForm" hx-post="/link-add" hx-target="main" hx-swap="innerHTML" hx-on--after-request="this.reset()" hx-indicator="#indicator" hx-disabled-elt="#linkAddSubmitButton" hx-vals="js:{created_at: Date.now()}"><label for="link-add">New link</label><input type="url" name="url" id="link-add" placeholder="https://picocss.com/docs/forms" aria-describedby="link-add-aria-description" required="" minlength="12"/><small id="link-add-aria-description">Write ot paste link with https protocol</small><button id="linkAddSubmitButton" type="submit">Add new link</button></form><form hx-post="/search" hx-target="#searchResult" hx-swap="innerHTML" hx-on--before-request="searchClearButton?.remove()" hx-on--after-request="this.reset()" hx-indicator="#indicator" hx-disabled-elt="#searchFormSubmitButton" role="search"><input type="search" name="search" id="search" placeholder="picocss" minlength="2" required="" autofocus=""/><button id="searchFormSubmitButton" type="submit">Search</button></form><section id="searchResult"></section><div style="display: flex; align-items: center;"><fieldset><legend>Select view</legend><label><span>Card </span><input name="viewState" type="checkbox" role="switch" style="margin-inline: .5em;" hx-post="/link-change-view" hx-target="main" hx-swap="innerHTML"/><span>Table</span></label></fieldset><span id="linksCounter" hx-swap-oob="outerHTML" style="width: 10ch; text-align: right;">30 pcs</span></div></nav></header><dialog id="settingsView"><article><header style="display: flex; align-items: center;"><h2>Settings</h2><button class="outline" style="margin-left: auto;" onclick="settingsView.toggleAttribute(&#39;open&#39;); document.body.classList.toggle(&#39;modal-is-open&#39;);history.back();" hx-post="/link-add-form-update" hx-target="#linkAddForm" hx-swap="outerHTML">Close</button></header><div><form hx-post="/user-update-name" hx-target="this" hx-swap="outerHTML" hx-on--after-request="this.reset()"><label for="userName">Write your name</label><input type="text" id="userName" name="userName" inputmode="text" minlength="1" required="" value="admin" placeholder="admin"/><input type="submit" value="Update name"/></form><a href="/user-data-download">Download your data</a></div><div><form hx-post="/tag-create" hx-target="#createTagResponse" hx-swap="afterbegin" hx-on--after-request="this.reset()"><fieldset><legend style="text-align: center; width: 100%; font-weight: bold;">Create new tag</legend><label for="inputTagName">Write tag name</label><input type="text" name="name" id="inputTagName" placeholder="web" inputmode="text" minlength="3" required="" autofocus=""/><label for="inputTagColor">Choose a color</label><input type="color" name="color" id="inputTagColor" required=""/><input type="submit" value="Create tag"/></fieldset></form><p id="createTagResponse" style="display: flex; flex-wrap: wrap; gap: 1rem; min-height: 2rem;"></p></div><footer></footer></article></dialog><main></main>"
`;

exports[`Page Dashboard view table 1`] = `
"<header><div style="display:flex;justify-content:space-between;align-items:center;margin-bottom:var(--pico-typography-spacing-vertical)"><h1 id="userGreeting" style="margin: 0;"><a href="/dashboard">Hello, admin!</a></h1><div style="display: flex; align-items: center; gap: 1rem;"><label><span>Dark</span><input type="checkbox" role="switch" style="margin-inline: .5em;" x-data="{
        dataTheme: undefined, 
        dataThemeValues: [&#39;dark&#39;, &#39;light&#39;],
        get dataThemeInputValue() {
            return &#39;dark&#39; === this.dataTheme ? false : true; 
        },
        setThemeCookie() {
            document.cookie = &#39;theme=&#39; + this.dataTheme + &#39;;SameSite=Strict;&#39;;
        },
        getThemeCookie() {
            const theme = document.cookie?.includes(&#39;theme&#39;) ? document.cookie.split(&#39;=&#39;).at(1) : undefined;
            return this.dataThemeValues.includes(theme) ? theme : undefined;
        },
        setTheme() {
            this.setThemeCookie();
            document.documentElement.setAttribute(&#39;data-theme&#39;, this.dataTheme);
        },
        toggleTheme() {
            this.dataTheme = &#39;dark&#39; === this.dataTheme ? &#39;light&#39; : &#39;dark&#39;;
            this.setTheme();
        },
            init() {
                const cookieTheme = this.getThemeCookie();
                if(cookieTheme) {
                    this.dataTheme = cookieTheme;
                    this.setTheme();
                    return;
                }
                this.dataTheme = window.matchMedia(&#39;(prefers-color-scheme: dark)&#39;).matches ? &#39;dark&#39; : &#39;light&#39;;
            },
    }" x-on:click="toggleTheme" x-model="dataThemeInputValue"/><span>Light</span></label><button class="outline secondary" onclick="settingsView.toggleAttribute(&#39;open&#39;); document.body.classList.toggle(&#39;modal-is-open&#39;);history.pushState({}, &#39;&#39;, &#39;/settings&#39;);">Settings</button><form action="/logout" method="post" style="margin: 0;"><button type="submit" class="outline" style="margin: 0;">Logout</button></form></div></div><nav style="display: grid; grid-template-columns: 1fr;"><form id="linkAddForm" hx-post="/link-add" hx-target="main" hx-swap="innerHTML" hx-on--after-request="this.reset()" hx-indicator="#indicator" hx-disabled-elt="#linkAddSubmitButton" hx-vals="js:{created_at: Date.now()}"><label for="link-add">New link</label><input type="url" name="url" id="link-add" placeholder="https://picocss.com/docs/forms" aria-describedby="link-add-aria-description" required="" minlength="12"/><small id="link-add-aria-description">Write ot paste link with https protocol</small><button id="linkAddSubmitButton" type="submit">Add new link</button></form><form hx-post="/search" hx-target="#searchResult" hx-swap="innerHTML" hx-on--before-request="searchClearButton?.remove()" hx-on--after-request="this.reset()" hx-indicator="#indicator" hx-disabled-elt="#searchFormSubmitButton" role="search"><input type="search" name="search" id="search" placeholder="picocss" minlength="2" required="" autofocus=""/><button id="searchFormSubmitButton" type="submit">Search</button></form><section id="searchResult"></section><div style="display: flex; align-items: center;"><fieldset><legend>Select view</legend><label><span>Card </span><input name="viewState" type="checkbox" role="switch" style="margin-inline: .5em;" checked="" hx-post="/link-change-view" hx-target="main" hx-swap="innerHTML"/><span>Table</span></label></fieldset><span id="linksCounter" hx-swap-oob="outerHTML" style="width: 10ch; text-align: right;">30 pcs</span></div></nav></header><dialog id="settingsView"><article><header style="display: flex; align-items: center;"><h2>Settings</h2><button class="outline" style="margin-left: auto;" onclick="settingsView.toggleAttribute(&#39;open&#39;); document.body.classList.toggle(&#39;modal-is-open&#39;);history.back();" hx-post="/link-add-form-update" hx-target="#linkAddForm" hx-swap="outerHTML">Close</button></header><div><form hx-post="/user-update-name" hx-target="this" hx-swap="outerHTML" hx-on--after-request="this.reset()"><label for="userName">Write your name</label><input type="text" id="userName" name="userName" inputmode="text" minlength="1" required="" value="admin" placeholder="admin"/><input type="submit" value="Update name"/></form><a href="/user-data-download">Download your data</a></div><div><form hx-post="/tag-create" hx-target="#createTagResponse" hx-swap="afterbegin" hx-on--after-request="this.reset()"><fieldset><legend style="text-align: center; width: 100%; font-weight: bold;">Create new tag</legend><label for="inputTagName">Write tag name</label><input type="text" name="name" id="inputTagName" placeholder="web" inputmode="text" minlength="3" required="" autofocus=""/><label for="inputTagColor">Choose a color</label><input type="color" name="color" id="inputTagColor" required=""/><input type="submit" value="Create tag"/></fieldset></form><p id="createTagResponse" style="display: flex; flex-wrap: wrap; gap: 1rem; min-height: 2rem;"></p></div><footer></footer></article></dialog><main></main>"
`;

exports[`Page Dashboard view table with tags 1`] = `
"<header><div style="display:flex;justify-content:space-between;align-items:center;margin-bottom:var(--pico-typography-spacing-vertical)"><h1 id="userGreeting" style="margin: 0;"><a href="/dashboard">Hello, admin!</a></h1><div style="display: flex; align-items: center; gap: 1rem;"><label><span>Dark</span><input type="checkbox" role="switch" style="margin-inline: .5em;" x-data="{
        dataTheme: undefined, 
        dataThemeValues: [&#39;dark&#39;, &#39;light&#39;],
        get dataThemeInputValue() {
            return &#39;dark&#39; === this.dataTheme ? false : true; 
        },
        setThemeCookie() {
            document.cookie = &#39;theme=&#39; + this.dataTheme + &#39;;SameSite=Strict;&#39;;
        },
        getThemeCookie() {
            const theme = document.cookie?.includes(&#39;theme&#39;) ? document.cookie.split(&#39;=&#39;).at(1) : undefined;
            return this.dataThemeValues.includes(theme) ? theme : undefined;
        },
        setTheme() {
            this.setThemeCookie();
            document.documentElement.setAttribute(&#39;data-theme&#39;, this.dataTheme);
        },
        toggleTheme() {
            this.dataTheme = &#39;dark&#39; === this.dataTheme ? &#39;light&#39; : &#39;dark&#39;;
            this.setTheme();
        },
            init() {
                const cookieTheme = this.getThemeCookie();
                if(cookieTheme) {
                    this.dataTheme = cookieTheme;
                    this.setTheme();
                    return;
                }
                this.dataTheme = window.matchMedia(&#39;(prefers-color-scheme: dark)&#39;).matches ? &#39;dark&#39; : &#39;light&#39;;
            },
    }" x-on:click="toggleTheme" x-model="dataThemeInputValue"/><span>Light</span></label><button class="outline secondary" onclick="settingsView.toggleAttribute(&#39;open&#39;); document.body.classList.toggle(&#39;modal-is-open&#39;);history.pushState({}, &#39;&#39;, &#39;/settings&#39;);">Settings</button><form action="/logout" method="post" style="margin: 0;"><button type="submit" class="outline" style="margin: 0;">Logout</button></form></div></div><nav style="display: grid; grid-template-columns: 1fr;"><form id="linkAddForm" hx-post="/link-add" hx-target="main" hx-swap="innerHTML" hx-on--after-request="this.reset()" hx-indicator="#indicator" hx-disabled-elt="#linkAddSubmitButton" hx-vals="js:{created_at: Date.now()}"><label for="link-add">New link</label><input type="url" name="url" id="link-add" placeholder="https://picocss.com/docs/forms" aria-describedby="link-add-aria-description" required="" minlength="12"/><small id="link-add-aria-description">Write ot paste link with https protocol</small><label for="tags">Select tags</label><select id="tags" name="tags" required="" aria-label="Select"><option disabled="" selected="" value="NaN">Select one:</option><option value="0">web</option></select><button id="linkAddSubmitButton" type="submit">Add new link</button></form><form hx-post="/search" hx-target="#searchResult" hx-swap="innerHTML" hx-on--before-request="searchClearButton?.remove()" hx-on--after-request="this.reset()" hx-indicator="#indicator" hx-disabled-elt="#searchFormSubmitButton" role="search"><input type="search" name="search" id="search" placeholder="picocss" minlength="2" required="" autofocus=""/><button id="searchFormSubmitButton" type="submit">Search</button></form><section id="searchResult"></section><div style="display: flex; align-items: center;"><fieldset><legend>Select view</legend><label><span>Card </span><input name="viewState" type="checkbox" role="switch" style="margin-inline: .5em;" checked="" hx-post="/link-change-view" hx-target="main" hx-swap="innerHTML"/><span>Table</span></label></fieldset><span id="linksCounter" hx-swap-oob="outerHTML" style="width: 10ch; text-align: right;">30 pcs</span></div></nav></header><dialog id="settingsView"><article><header style="display: flex; align-items: center;"><h2>Settings</h2><button class="outline" style="margin-left: auto;" onclick="settingsView.toggleAttribute(&#39;open&#39;); document.body.classList.toggle(&#39;modal-is-open&#39;);history.back();" hx-post="/link-add-form-update" hx-target="#linkAddForm" hx-swap="outerHTML">Close</button></header><div><form hx-post="/user-update-name" hx-target="this" hx-swap="outerHTML" hx-on--after-request="this.reset()"><label for="userName">Write your name</label><input type="text" id="userName" name="userName" inputmode="text" minlength="1" required="" value="admin" placeholder="admin"/><input type="submit" value="Update name"/></form><a href="/user-data-download">Download your data</a></div><div><form hx-post="/tag-create" hx-target="#createTagResponse" hx-swap="afterbegin" hx-on--after-request="this.reset()"><fieldset><legend style="text-align: center; width: 100%; font-weight: bold;">Create new tag</legend><label for="inputTagName">Write tag name</label><input type="text" name="name" id="inputTagName" placeholder="web" inputmode="text" minlength="3" required="" autofocus=""/><label for="inputTagColor">Choose a color</label><input type="color" name="color" id="inputTagColor" required=""/><input type="submit" value="Create tag"/></fieldset></form><p id="createTagResponse" style="display: flex; flex-wrap: wrap; gap: 1rem; min-height: 2rem;"><span style="color:blue;background-color:blue30;;border:solid 1px blue;border-radius:1rem;padding:0.125rem 0.5rem 0.25rem;display:flex;align-items:center">web <button type="button" class="outline" style="color:blue;border-color:blue;padding:0;;border-radius:100%;;width:3ch;;aspect-ratio:1;;font-size:0.75rem;;margin:0;;margin-left:0.5rem;" name="tagId" value="0" hx-post="/tag-delete" hx-swap="delete" hx-target="closest span">X</button></span></p></div><footer></footer></article></dialog><main></main>"
`;
